packages/api/cms-api/src/page-tree/page-tree-read-api.ts

```diff
@@ -86,6 +86,10 @@ export function createReadApi(
 
             nodes = filterPreloadedNodes(nodes, where);
 
+            if (sort && sort[0].field === "pos" && sort[0].direction === "ASC") {
+                sort.shift();
```

Comment:
Can you explain in more detail why it is not correct for multiple field sort, what exactly is buggy below and what you mean by "sort should be reversed"?

My thought in these two lines was to remove the first element of the sort array if this element is an ascending sort by position, because that is the default sort. The remaining elements are then applied for the sorting. Further occurrences of sorting by position cannot be removed because the sorting would then no longer be correct.

Reply:
when applying `sort = [{ field: "pos", direction: "ASC"}, { field: "name", direction: "ASC'}]` you are sorting by name.

And with `sort = [{ field: "name", direction: "ASC"}, { field: "updatedAt", direction: "ASC'}]` this is also wrong (not related to your changes) I think you need sort.reverse().forEach or so

Reply:
it would really be helpful to have unit tests for that code!

Reply:
Removing the first sort criteria when it is "pos, ascending" is not correct, as the nodes would then be sorted by the second criteria instead, ignoring the position altogether. This optimization only works when "pos, ascending" is the only criteria to sort by. I've added the optimization here: 9cb467fc9624afa0c4d0071be1cb2040b834ba96